src/lib/utils/cache.ts

```diff
@@ -18,22 +18,56 @@ export async function generateKey(...valuesToConsider: string[]): Promise<string
     compilerCliVersion = (await ngCompilerCli()).VERSION.full;
   }
 
-  return createHash('sha1')
+  return createHash('sha256')
     .update(ngPackagrVersion)
     .update(compilerCliVersion)
     .update(valuesToConsider.join(':'))
     .digest('hex');
 }
 
-export async function readCacheEntry(cachePath: string, key: string): Promise<any> {
-  const entry = await cacache.get.info(cachePath, key);
-  if (entry) {
-    return JSON.parse(await readFile(entry.path, 'utf8'));
+async function ensureCacheDirExists(cachePath: string): Promise<void> {
+  try {
+    await mkdir(cachePath, { recursive: true });
+  } catch (err) {
+    if (err.code !== 'EEXIST') {
+      throw err;
+    }
   }
+}
+
+export async function readCacheEntry(cachePath: string, key: string): Promise<any> {
+  const filePath = join(cachePath, key);
+
+  try {
+    const data = await readFile(filePath, 'utf8');
+
+    return JSON.parse(data, (_key, value) => {
+      if (typeof value === 'string' && value.endsWith('n')) {
+        const numPart = value.slice(0, -1);
 
-  return undefined;
+        // Only convert valid numeric strings
+        return isNaN(numPart as any) ? value : BigInt(numPart);
+      }
+
+      return value;
+    });
+  } catch (err) {
+    if (err.code === 'ENOENT') {
+      // File does not exist (cache miss)
+      return undefined;
+    }
+
+    throw err;
+  }
 }
 
-export async function saveCacheEntry(cachePath: string, key: string, content: string): Promise<void> {
-  await cacache.put(cachePath, key, content);
+export async function saveCacheEntry(cachePath: string, key: string, content: any): Promise<void> {
+  const filePath = join(cachePath, key);
+
+  // Ensure the cache directory exists
+  await ensureCacheDirExists(cachePath);
+
+  const data = JSON.stringify(content, (_key, value) => (typeof value === 'bigint' ? value.toString() + 'n' : value));
```

Comment:
WDYT of making this a bit more explicit, like `%BigInt(${value})` to reduce the chance of stripping off an `n` suffix of a string that happens to be `(\d+)n`

Reply:
Good idea.